### PR TITLE
Accept expr wherever typeExpr appears in the grammar.

### PR DESCRIPTION
**Pull request by dabrahams**

### CODE REVIEW

File: Tests/ValTests/ParserTests.swift [hunk 1]
```diff
@@ -1648,6 +1648,12 @@ final class ParserTests: XCTestCase {
     XCTAssertEqual(list.count, 2)
   }
 
+  func testTypeExprUnificationWithExpr() throws {
+    let input = SourceFile(contents: "<T, @value 40 + two()>")
```
Comment (kyouko-taiga) [inline Tests/ValTests/ParserTests.swift:1652]: Do tests succeed without the `@value`?

**Review by kyouko-taiga — CHANGES_REQUESTED** [review]
Could you create a test that doesn't have `@value` in front of something that cannot parse as a type expression.

**Comment by dabrahams** [comment]
Yes!

**Comment by dabrahams** [comment]
I can, but it's not a very specific request. Is this test supposed to fail or succeed? What grammar symbol am I trying to parse.

**Comment by kyouko-taiga** [comment]
The test is supposed to succeed. I believe I suggested:

```
func testStaticArgumentList() throws {
  let input = SourceFile(contents: "<T, 40 + two()>")
  let list = try XCTUnwrap(try apply(Parser.staticArgumentList, on: input).element)
  XCTAssertEqual(list.count, 2)
}
```

The goal is to determine whether the parser can accept an expression where it used to only parse type expressions without any "hint".

**Comment by kyouko-taiga** [comment]
OK, good enough for now I guess.

**Review by kyouko-taiga — APPROVED** [review]

**Pull request merged**